rpc/errors.go

```diff
@@ -18,6 +18,11 @@ package rpc
 
 import "fmt"
 
+var (
+	// ErrConnectionClosed is thrown when the connection is closed
+	ErrConnectionClosed = new(connectionClosedError)
```

Comment:
Maybe it would be nice to have `func newError(code int, format string, args ...interface{}) RPCError`. It could be used to remove more error types which only exist to satisfy the interface.